extensions/amp-access/0.1/amp-access.js

```diff
@@ -233,6 +253,84 @@ export class AccessService {
   }
 
   /**
+   * @return {!Promise}
+   * @private
+   */
+  scheduleView_() {
+    return this.whenViewed_().then(this.registerView_.bind(this), reason => {
+      // Ignore - view has been canceled.
+      log.fine(TAG, 'view cancelled:', reason);
+    });
+  }
+
+  /**
+   * @return {!Promise}
```

Comment:
Nice, maybe add doc:
The promise will be resolved when a view of this document has occurred. It will be rejected if the current render should not be counted as a view.